sig-and-wg/sig-core/proposals/component-folder-consolidation.md

```diff
@@ -0,0 +1,118 @@
+
+Kubernetes API specific:
+- **controller** - is a [Kubernetes Controller](https://kubernetes.io/docs/concepts/extend-kubernetes/extend-cluster/) named by the primary kubernetes resource which it is controlling, like _api-controller_.
+- **operator** - is a [Kubernetes Operator](https://coreos.com/operators/) named by the module it is operating, like _application-operator_.
+- **job** - is a [Kubernetes Job](https://kubernetes.io/docs/tasks/job/), performing a task once or periodic, named by the task it is doing, like _istio-patch-job_
+- **proxy** - is proxying an existing component usually introducing a security model for the proxied component, named by the component, like _api-server-proxy_.
+
+Business API specific:
+- **service** - is serving an HTTP/S-based API, usually exposed securely to the public, named by the domain and API it is serving, like _application-connector-service_.
+- **broker** - is implementing the OpenServiceBroker API, named by its provider it is integrating into, like _azure-broker_.
+
+### Concrete Action Items
+
+- Have a readme in the _components_ folder explaining the naming convention for components
+- Link the readme in the developer guide
+- All folders will stay in the root components folder but will be renamed according to the following table:
+
+|type| old folder | new folder | action required |
+|------------|------------|-----------------|----|
+|**-broker**|application-broker|application-broker| - |
+| |helm-broker`*`|helm-broker| - |
+|**-controller**|api-controller|api-controller| - |
+| |binding-usage-controller|service-binding-usage-controller| yes |
+| |connection-token-handler|request-token-controller| yes|
+| |namespace-controller|namespace-controller|no|
+| |idppreset|idppreset-controller|yes|
+| |event-bus`**`/event-bus-sv|event-activation-controller|yes|
+| |event-bus`**`/event-bus-push|subscription-controller|yes|
+|**-operator**|application-operator|application-operator| - |
+| |installer|kyma-operator| yes|
+|**-service**|configurations-generator|iam-kubeconfig-service|yes|
+| |connector-service|application-connector-service|yes|
+| |application-registry| application-registration-service|yes|
+| |event-service|application-event-service| yes|
+| |application-proxy|application-proxy-service|yes|
+| |event-bus`**`/event-bus-publish|event-publish-service|yes|
+| |ui-api-layer|console-backend-service|yes|
+|**-proxy**|apiserver-proxy|apiserver-proxy| - |
+| |k8s-dashboard-proxy|k8s-dashboard-proxy| - |
+|**-job**|istio-kyma-patch|istio-patch-job|yes|
+
+`*` should be splitted into the helm-broker and tooling for bundle repositories outside of kyma repo
+
+`**` should be splitted into the different components, the common parts not specific to eventing should be moved into the kyma common parts, otherwise they should be duplicated
```

Comment:
```suggestion
`**` should be split into the different components, the common parts not specific to eventing should be moved to the Kyma common parts to avoid duplication
```